lib/ansible/modules/cloud/amazon/lambda.py

```diff
@@ -430,6 +432,9 @@ def main():
 
     # Function doesn't exists, create new Lambda function
     elif state == 'present':
+        if not role_arn:
+            module.fail_json(msg='A role is required for Lambda function creation')
```

Comment:
The reason I didn't do that is because role_arn doesn't need to be required with state == present, only if the function doesn't exist. The if statement above this elif (also checking if state is present) doesn't need a role_arn.